Neos.ContentRepository.Core/Classes/SharedModel/Node/NodeAggregateId.php

```diff
@@ -50,6 +50,23 @@ public static function fromString(string $value): self
         return new self($value);
     }
 
+    /**
+     * Calculates a deterministic nodeAggregateId.
+     *
+     * Used for determining the nodeAggregateIds of tethered nodes.
+     *
+     * e.g. in case you want to get the main content collection node aggregate id:
+     *  NodeAggregateId::fromParentNodeAggregateIdAndNodeName($parentNodeAggregateId, NodeName::fromString('main'))
+     *
+     */
+    public static function fromParentNodeAggregateIdAndNodeName(NodeAggregateId $parentNodeAggregateId, NodeName $tetheredNodeName): self
+    {
+        $hex = md5($parentNodeAggregateId->value . '-' . $tetheredNodeName->value);
+        return new self(
+            substr($hex, 0, 8) . '-' . substr($hex, 8, 4) . '-' . substr($hex, 12, 4) . '-' . substr($hex, 16, 4) . '-' . substr($hex, 20, 12)
+        );
```

Comment:
to discuss: does this have to be a UUID pattern?

Reply:
to follow up on this:
node ids used to be UUIDs and that is the reason for this algorithm.
But now they can be anything basically (as long as they match the regex `/^([a-z0-9\-]{1,64})$/`).

I would suggest to just do `$parentNodeAggregateId->value . '-' . $tetheredNodeName->value` really (with some deterministic changes to satisfy the regex)